app/src/main/java/com/practicum/playlistmaker/player/presentation/MediaPlayerViewModel.kt

```diff
@@ -50,15 +34,26 @@ class MediaPlayerViewModel(private val mediaPlayerInteractor: MediaPlayerInterac
             mediaPlayerInteractor.preparePlayer(trackPreviewUrl)
             isCreated = true
         }
-        preparePlayerRunnable.let { handler.post(it) }
+        prepareJob = viewModelScope.launch {
+            while (!mediaPlayerInteractor.isMediaPlayerPrepared) {
+                delay(CHECK_PREPARE_PLAYER_DELAY)
+            }
+            renderState(PlayerState.Prepared(mediaPlayerInteractor.getTrackPosition()))
+        }
     }
 
     fun playbackControl() {
         when (stateLiveData.value) {
             is PlayerState.Prepared, PlayerState.Pause, PlayerState.Complete -> {
                 mediaPlayerInteractor.startPlayer()
                 renderState(PlayerState.Playing)
-                timerUpdater.let { handler.post(it) }
+                timerJob = viewModelScope.launch {
```

Comment:
При запуске новой джобы лучше на всякий случай всегда отменять старую:
```suggestion
                timerJob?.cancel()
                timerJob = viewModelScope.launch {
```